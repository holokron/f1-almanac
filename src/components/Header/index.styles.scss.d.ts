export const headerRow: string;
export const headerSpan: string;
export const headerLogo: string;
