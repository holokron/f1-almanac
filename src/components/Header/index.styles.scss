.headerRow {
  color: white;
  font-weight: bold;
}

.headerSpan {
  margin-left: 1rem;
}

.headerLogo {
  width: 64px;
  height: 50px;
}
