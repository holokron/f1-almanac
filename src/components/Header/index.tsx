import React, { ReactElement } from "react"
import { Row } from "antd"
import styles from "./index.module.css"
import Logo from "./logo.svg"
import { Link } from "gatsby"

interface HeaderProps {
  title: string
}

function Header({ title }: HeaderProps): ReactElement<HeaderProps> {
  return (
    <Link to="/">
      <Row
        className={styles.headerRow}
        type="flex"
        align="middle"
        justify="center"
      >
        <Logo className={styles.headerLogo} />{" "}
        <span className={styles.headerSpan}>{title}</span>
      </Row>
    </Link>
  )
}

export default React.memo(Header)
