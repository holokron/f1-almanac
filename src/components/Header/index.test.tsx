import React from "react"
import renderer from "react-test-renderer"
import Header from "."

describe("components/Header", (): void => {
  it.each([["Foo Bar"]])(
    "title: %p",
    (title: string): void => {
      const tree = renderer.create(<Header title={title} />)

      expect(tree.toJSON()).toMatchSnapshot()
    },
  )
})
