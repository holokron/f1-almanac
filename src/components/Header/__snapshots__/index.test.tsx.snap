// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header title: "Foo Bar" 1`] = `
<a
  aria-current="page"
  className=""
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={Object {}}
>
  <div
    className="ant-row-flex ant-row-flex-center ant-row-flex-middle headerRow"
  >
    <test-file-stub
      className="headerLogo"
    />
     
    <span
      className="headerSpan"
    >
      Foo Bar
    </span>
  </div>
</a>
`;
